service/history/workflow/update/registry.go

```diff
@@ -352,3 +343,18 @@ func (r *registry) findLocked(ctx context.Context, id string) (*Update, bool) {
 		withInstrumentation(&r.instrumentation),
 	), true
 }
+
+// enumerate returns a slice of all updates in the registry for which the
+// provided filter function returns true. The registry is locked for reading
+// while enumerating over the map. Resulted slice can be iterated w/o holding a lock.
+func (r *registry) enumerate(filter func(u *Update) bool) []*Update {
```

Comment:
The standard name is filter for the function and predicate for the argument. Then you will not need to explain meaning in the docstring.